dotenv/mod_test.ts

```diff
@@ -643,3 +644,55 @@ Deno.test("stringify", async (t) => {
       ),
   );
 });
+
+Deno.test("use restrictEnvAccessTo to restrict lookup of Env variables to certain vars. Those vars can be granted read permissions now separately.", async () => {
+  const { stdout } = await Deno.spawn(Deno.execPath(), {
+    args: [
+      "run",
+      "--allow-read",
+      "--allow-env=GREETING",
+      path.join(testdataDir, "./app_with_restricted_env_access.ts"),
+    ],
+    cwd: testdataDir,
+  });
+
+  const decoder = new TextDecoder();
+  const conf = JSON.parse(decoder.decode(stdout).trim());
+
+  assertEquals(conf.GREETING, "hello world", "fetches .env by default");
+  assertEquals(conf.DEFAULT1, "Some Default", "default value loaded");
+});
+
+Deno.test("use restrictEnvAccessTo via configSync to restrict lookup of Env variables to certain vars.", async () => {
+  const { stdout } = await Deno.spawn(Deno.execPath(), {
+    args: [
+      "run",
+      "--allow-read",
+      "--allow-env=GREETING",
+      path.join(testdataDir, "./app_with_restricted_env_access_sync.ts"),
+    ],
+    cwd: testdataDir,
+  });
+
+  const decoder = new TextDecoder();
+  const conf = JSON.parse(decoder.decode(stdout).trim());
+
+  assertEquals(conf.GREETING, "hello world", "fetches .env by default");
+  assertEquals(conf.DEFAULT1, "Some Default", "default value loaded");
+});
+
+Deno.test("use of restrictEnvAccessTo for an Env var, without granting read permissions still fails", async () => {
```

Comment:
```suggestion
Deno.test("use of restrictEnvAccessTo for an Env var, without granting env permissions still fails", async () => {
```

Reply:
The delay is very ok, happy that anyone  has time for that at all!

The suggested change is of course welcome 👍